src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -312,6 +312,13 @@ private TCommandResult ProcessResponse(ConnectionId connectionId, CommandMessage
                     if (materializedDocument.TryGetValue("errmsg", out errmsgBsonValue) && errmsgBsonValue.IsString)
                     {
                         var errmsg = errmsgBsonValue.ToString();
+                        if (materializedDocument.TryGetValue("code", out var errorCode)
+                            && errorCode.AsInt32 == 20
+                            && errmsg.StartsWith("Transaction numbers"))
+                        {
+                            errmsg = "This MongoDB deployment does not support retryable writes. "
+                                     + "Please add retryWrites=false to your connection string.";
+                        }
```

Comment:
It might be better to create a new exception with the original (unmodified) exception as an InnerException rather than hijack the current exception and lose a bit of information from the original server message.

Reply:
Fair. Done!

Reply:
Hm, is throwing a different type of exception a breaking change to the API? @rstam ? Seems like it could be

Reply:
I could expand `MongoCommandException` so that it takes an inner exception param, and that should save us I think

Reply:
Can't we still throw a MongoCommandException?

Reply:
Did the above, LMK what you think!

Reply:
I'm fine with MongoCommandException having a new constructor for an inner exception.